encord/http/querier.py

```diff
@@ -135,7 +136,18 @@ def execute(self, request, enable_logging=True):
         timeouts = (request.connect_timeout, request.timeout)
 
         try:
-            res = session.send(req, timeout=timeouts)
+
+            @retry_on_network_errors
```

Comment:
Mostly a question for my own education: why can't we add the parameters to the decorator here rather than the caller? Do these get "too captured" and not change between calls to the outer function here?

Reply:
It depends. 
* If it is a function that is being created inside this other function here, we could also have the arguments in the decorator.
* If the decorator is used for a function in "global scope" it won't work cause we cannot dynamically set the arguments as far as I found.